lib/samlp.js

```diff
@@ -211,19 +211,20 @@ module.exports.auth = function(options) {
 
   return function (req, res, next) {
     getSamlRequest(req.query.SAMLRequest || req.body.SAMLRequest, function(err, samlRequestDom) {
+      var audience = options.audience;
       if (samlRequestDom) {
         var issuer = xpath.select("//*[local-name(.)='Issuer' and namespace-uri(.)='urn:oasis:names:tc:SAML:2.0:assertion']/text()", samlRequestDom);
-        if (issuer && issuer.length > 0) options.audience = options.audience || issuer[0].textContent;
+        if (issuer && issuer.length > 0) audience = issuer[0].textContent || audience;
```

Comment:
`options.audience` was introduced to override the value from SAMLRequest. See: https://github.com/auth0/node-samlp/commit/63f9a76c2d1d2ba60607b72d085a50fad6045a0a